### PR TITLE
Update set commands

### PR DESCRIPTION
**Pull request by soltysh**
/assign @deads2k @juanvallejo

### CODE REVIEW

File: pkg/oc/cli/cmd/set/buildsecret.go [hunk 1]
```diff
@@ -218,85 +190,103 @@ func (o *BuildSecretOptions) Validate() error {
...
-		kcmdutil.PrintSuccess(o.ShortOutput, o.Out, info.Object, false, "updated")
-	}
-	if len(errs) > 0 {
-		return errors.NewAggregate(errs)
-	}
-	return nil
-}
+		actual, err := o.Client.Resource(info.Mapping.Resource).Namespace(info.Namespace).Patch(info.Name, types.StrategicMergePatchType, patch.Patch)
+		if err != nil {
+			allErrs = append(allErrs, fmt.Errorf("fialed to patch secret  %v", err))
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/buildsecret.go]: s/fialed/failed

File: pkg/oc/cli/cmd/set/buildhook.go [hunk 2]
```diff
@@ -2,21 +2,20 @@ package set
...
 	"k8s.io/apimachinery/pkg/api/meta"
-	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/client-go/dynamic"
 	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
 	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+	"k8s.io/kubernetes/pkg/printers"
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/buildhook.go]: I thought we had switched to using the new `pkg/genericclioptions/printers` pkg?

File: pkg/oc/cli/cmd/set/buildhook.go [hunk 3]
```diff
@@ -235,39 +219,44 @@ func (o *BuildHookOptions) Run() error {
...
-	if len(o.Output) > 0 || o.Local || kcmdutil.GetDryRunFlag(o.Cmd) {
-		return o.PrintObject(infos)
-	}
-
-	failed := false
+	allErrs := []error{}
 	for _, patch := range patches {
 		info := patch.Info
+		name := info.Name
+		if info.Mapping != nil {
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/buildhook.go]: This fixes the panic we were seeing with --local, right?

File: pkg/oc/cli/cmd/set/buildhook.go [hunk 4]
```diff
@@ -235,39 +219,44 @@ func (o *BuildHookOptions) Run() error {
...
+		}
 		if patch.Err != nil {
-			fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+			allErrs = append(allErrs, fmt.Errorf("error: %s %v\n", name, patch.Err))
 			continue
 		}
 
 		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
-			fmt.Fprintf(o.Err, "info: %s %q was not changed\n", info.Mapping.Resource, info.Name)
+			fmt.Fprintf(o.ErrOut, "info: %s was not changed\n", name)
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/buildhook.go]: I thought we were getting rid of this message?

File: pkg/oc/cli/cmd/set/buildsecret.go [hunk 5]
```diff
@@ -2,24 +2,22 @@ package set
...
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/errors"
-	"k8s.io/kubernetes/pkg/api/legacyscheme"
+	"k8s.io/client-go/dynamic"
 	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
 	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+	"k8s.io/kubernetes/pkg/printers"
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/buildsecret.go]: genericclioptions/printers?

File: pkg/oc/cli/cmd/set/deploymenthook.go [hunk 6]
```diff
@@ -237,66 +222,84 @@ func (o *DeploymentHookOptions) Validate() error {
...
 
-	if singleItemImplied && len(patches) == 0 {
-		return fmt.Errorf("%s/%s is not a deployment config or does not have an applicable strategy", infos[0].Mapping.Resource, infos[0].Name)
-	}
-
-	if len(o.Output) > 0 || o.Local || kcmdutil.GetDryRunFlag(o.Cmd) {
-		return o.PrintObject(infos)
-	}
+	// if singleItemImplied && len(patches) == 0 {
+	// 	return fmt.Errorf("%s/%s is not a deployment config or does not have an applicable strategy", infos[0].Mapping.Resource, infos[0].Name)
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/deploymenthook.go]: Is this leftover, or are we enabling this error message in the future?

File: pkg/oc/cli/cmd/set/env.go [hunk 7]
```diff
@@ -81,81 +84,78 @@ var (
...
+
+	genericclioptions.IOStreams
+	resource.FilenameOptions
+}
 
-	Mapper meta.RESTMapper
-	Client dynamic.Interface
+func NewEnvOptions(streams genericclioptions.IOStreams) *EnvOptions {
+	return &EnvOptions{
+		PrintFlags: genericclioptions.NewPrintFlags("updated"),
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/env.go]: `.WithTypeSetter(oscheme.PrintingInternalScheme)`?

File: pkg/oc/cli/cmd/set/probe.go [hunk 8]
```diff
@@ -22,6 +21,7 @@ import (
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions"
 	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
 	"k8s.io/kubernetes/pkg/kubectl/polymorphichelpers"
+	"k8s.io/kubernetes/pkg/printers"
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/probe.go]: genericclioptions/printers

File: pkg/oc/cli/cmd/set/probe.go [hunk 9]
```diff
@@ -116,104 +108,90 @@ type ProbeOptions struct {
...
+	cmd.Flags().BoolVar(&o.All, "all", o.All, "If true, select all resources in the namespace of the specified resource types")
+	cmd.Flags().BoolVar(&o.Remove, "remove", o.Remove, "If true, remove the specified probe(s).")
+	cmd.Flags().BoolVar(&o.Readiness, "readiness", o.Readiness, "Set or remove a readiness probe to indicate when this container should receive traffic")
+	cmd.Flags().BoolVar(&o.Liveness, "liveness", o.Liveness, "Set or remove a liveness probe to verify this container is running")
+	cmd.Flags().BoolVar(&o.Local, "local", o.Local, "If true, set image will NOT contact api-server but run locally.")
+	cmd.Flags().StringVar(&o.OpenTCPSocket, "open-tcp", o.OpenTCPSocket, "A port number or port name to attempt to open via TCP.")
+	cmd.Flags().StringVar(&o.HTTPGet, "get-url", o.HTTPGet, "A URL to perform an HTTP GET on (you can omit the host, have a string port, or omit the scheme.")
+
+	o.InitialDelaySeconds = cmd.Flags().Int("initial-delay-seconds", 0, "The time in seconds to wait before the probe begins checking")
+	o.SuccessThreshold = cmd.Flags().Int("success-threshold", 0, "The number of successes required before the probe is considered successful")
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/set/probe.go]: maybe we could switch these to IntVars?

File: pkg/oc/cli/cmd/set/buildhook.go (hunk 3 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/set/buildhook.go]: Yup.

File: pkg/oc/cli/cmd/set/buildhook.go (hunk 4 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/set/buildhook.go]: Turned into `glog.V(1).Infof("info: %s was not changed\n", name)`

File: pkg/oc/cli/cmd/set/deploymenthook.go (hunk 6 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/set/deploymenthook.go]: Yeah, just fixed.

File: pkg/oc/cli/cmd/set/env.go (hunk 7 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/set/env.go]: Updated.

File: pkg/oc/cli/cmd/set/probe.go (hunk 9 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/set/probe.go]: There are pointer to values, I'm guessing to to differentiate between set and not, although there's `.Changed` paramter on each flag. But I didn't want to add more to this PR. We can solve it later ;)

**Comment by soltysh** [comment]
Comments addressed, ptal.

File: pkg/oc/cli/cmd/set/buildhook.go (hunk 3 above)
Comment (juanvallejo) [reply pkg/oc/cli/cmd/set/buildhook.go]: Per @liggitt's comment [here](https://github.com/kubernetes/kubernetes/pull/65636#discussion_r199308079), I am wondering if we would would want to do the same in Origin? That is, default to showing lower-case kind.group (with `.group` being included when non-empty) when `info.Mapping` is nil?

**Comment by juanvallejo** [comment]
One comment, otherwise tag at will

File: pkg/oc/cli/cmd/set/buildhook.go (hunk 3 above)
Comment (soltysh) [reply pkg/oc/cli/cmd/set/buildhook.go]: I've created `GetObjectName` which I'm guessing you can just add upstream, it does exactly what Jordan asked for.

**Comment by soltysh** [comment]
I'll wait for #20083 to merge and will tag this one.

**Comment by soltysh** [comment]
Approving based on previous comments and the included commit from #20083

**Comment by soltysh** [comment]
Rebase only, re-applying the label back.

**Pull request merged**